<launch>

  <arg name="rgb_optical_frame"           default="left_camera_color_optical_frame"/>
  <arg name="rviz"           default="true"/>

  <node type="pcl_processing" pkg="perception_playground"
        name="pcl_processing" args="$(arg rgb_optical_frame)">
    <remap from="~pointcloud" to="/seal_cameras/left_camera/depth/color/points"/>
    <remap from="~camera_info" to="/seal_cameras/left_camera/color/camera_info"/>
    <param name="z_max" type="double" value="0.33" />
    <!-- TODO what is difference between private param and regular param and what is best practice? -->
    <!-- <param name="fixed_frame" type="string" value="base_frame" /> -->
  </node>

  <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz" required="true" args="-d $(find perception_playground)/cfg/cfg1.rviz"/>

</launch>